features/standards/mag/structure/02_headings.feature

```diff
@@ -150,7 +162,7 @@ Feature: Headings
     When I test the "Structure: Headings: Headings must be in ascending order" standard
     Then it passes with the warning:
       """
-      First heading was not a main heading: /html/body/h3
+      First heading was not a level 1 or level 2 heading: /html/body/h3
```

Comment:
This is not in keeping with the guidance. The requirement is for only one `<h1>` per page, and a hierarchical heading structure following that `<h1>`. There is no specific guidance before the `<h1>` in regard heading levels. Ideally there would be no headings before the `<h1>`, and I would certainly wish to encourage `<h1>` to be the first heading within the main content container.

Reply:
> There is no specific guidance before the <h1> in regard heading levels

Just to double-check that: I can get on board with the h1 not being the _first_ heading in the page, keeping a h2 for Accessibility links and saving the h1 for the actual page content. That makes sense to me.

But if we were to arbitrarily use a h3, h4, h5 or h6 as the first heading in the page, that stops making sense! 

Is that not worth scanning for and warning against?

Reply:
I also agree with @ChrisBAshton.

However, it's true that there isn't really any guidance on this, and the passage in the HTML specification for sections and headings isn't even implemented in browsers (referring to the 'outline algorithm').

In an ideal world, the main article/content heading be determined by the fact that it's the top-level headline in the `main` content, rather than the heading level being `h1`; and the `h1` would always be the title of the website (e.g. BBC).

There's a big discussion around this at the WHATWG; see whatwg/html#3499

But it's going to be years since we have a definitive resolution to this issue. Maybe recommending hierarchy before the `h1` is the best compromise?

Reply:
I understand where you're coming from @ChrisBAshton but the one errant subheading in Orb could be an `<h4>` rather than an `<h2>` and that should make no difference. The guidance isn't permitting headings before the `<h1>` but the test doesn't have to fail if there is a preceding subheading, so long as the heading structure after the `<h1>` is hierarchical.

And this isn't trying to set any kind of precedent, as the WCAG guideline is different again in its requirements. This is simply dealing with the BBC situation and guideline.

Reply:
> I understand where you're coming from ChrisBAshton but the one errant subheading in Orb could be an <h4> rather than an <h2> and that should make no difference.
My opinion is that we should encourage users of the bbc-a11y tool to use the ignore functionality in cases like that, rather than changing the logic. (But in the case of Orbit, it always seems to stick to using `h2`.)

Actually, my opinion is changing on this. The W3C guidelines on this seem contradictory or ambiguous in terms of how strict we need to be with the heading hierarchy. If we were to change the logic so that we don't test the heading hierarchy before the `h1` element, does this mean we (theoretically) also aren't interested in testing any headings in the footer of the page?